src/platform/icelake/include/platform/memory.h

```diff
@@ -265,7 +265,7 @@
 /* text and data share the same HP L2 SRAM on Icelake */
 #define SOF_TEXT_START		0xBE040400
 #define SOF_TEXT_BASE		(SOF_TEXT_START)
-#define SOF_TEXT_MIN_SIZE	(0x1B000 - 0x400)
+#define SOF_TEXT_MIN_SIZE	(0x1B000 - 0x400 + 0x1000)
```

Comment:
Every time we add a component, which may or may not be selected by Kconfigs, we manually tweak the memory sizes. Doesn't look scalable to me.

Reply:
I'm not sure what you mean, but I think that 0x1000 should be at least define like other components that is set to 0x1000 if CONFIG_SELECTOR is here or 0x0. TBH we should not care about these TEXT sizes at all and it should just take as least as possible, calculated by tools not humans/defines.

Also when we continue having size per component you we will waste space - because we have to add size in pages - so for example component that takes 100 bytes have to take 1 page instead of just 100 bytes and when you have 2 components like that you will waste 2 pages with these defines etc...

Reply:
I second @jajanusz regarding long term approach , for now I've added the selector component support in memory.h same as it is done for SRC for example.

Reply:
@lbetlej @jajanusz yes long term is to remove, I cant quite remember why it was initially needed especially when IRAM/DRAM is separated physically on BYT anyway.

Reply:
> @lbetlej @jajanusz yes long term is to remove, I cant quite remember why it was initially needed especially when IRAM/DRAM is separated physically on BYT anyway.

Also it looks like it works only on APL because it's in base_module.c for apl and for others it's 0 and rimage can handle it

Reply:
That is right, almost every section that targets hpsram memory area (text/rodata/data/bss+heaps) may be packed into a single big section defined in MEMORY{} (see my req for changes in #991). It easy and quick to change and seems to be a short term must-have for apl since there are a lot of gaps hard to control manually.